pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java

```diff
@@ -28,6 +28,7 @@
 import java.util.Set;
 import java.util.TreeMap;
 import java.util.function.BiFunction;
+import org.apache.pinot.common.exception.QueryException;
```

Comment:
One high level comment on this class (and I should given this during previous PR) -- as part of Javadoc, it would be very helpful to summarize the modes and their combinations and purpose.

Reply:
Added it among the if else for all the 4 modes.